multy_test/test_properties.cpp

```diff
@@ -0,0 +1,555 @@
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ("0", str);
+}
+
+GTEST_TEST(PropertiesTestInvalidSetDate, properties_set_amount_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    std::string str = "0";
+    int32_t int_value = 1;
+    Amount amount(2);
+
+    const unsigned char data_4_vals[] = {1U, 2U, 3U, 4U};
+    BinaryDataPtr binaty_data_4_property(make_clone(BinaryData{data_4_vals, 4}));
+
+    AccountPtr account;
+    error.reset(make_account(
+                        CURRENCY_BITCOIN,
+                        "cQeGKosJjWPn9GkB7QmvmotmBbVg1hm8UjdN6yLXEWZ5HAcRwam7",
+                        reset_sp(account)));
+
+    PublicKeyPtr  pub_key_property =  account->get_public_key();
+
+    properties.bind_property("v", &amount);
+
+    error.reset(properties_set_int32_value(&properties, "v", int_value));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", amount.get_value().c_str());
+
+    error.reset(properties_set_string_value(&properties, "v", str.c_str()));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", amount.get_value().c_str());
+
+    error.reset(properties_set_binary_data_value(&properties, "v", binaty_data_4_property.get()));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", amount.get_value().c_str());
+
+    error.reset(properties_set_public_key_value(&properties, "v",  pub_key_property.get()));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", amount.get_value().c_str());
+}
+
+GTEST_TEST(PropertiesTestInvalidSetDate, properties_set_binary_data_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    std::string str = "0";
+    int32_t int_value = 1;
+    Amount amount(2);
+
+    const unsigned char data_4_vals[] = {1U, 2U, 3U, 4U};
+    const BinaryData reference_value{data_4_vals, 4};
+    BinaryDataPtr binaty_data_4_property(make_clone(BinaryData{data_4_vals, 4}));
+
+    AccountPtr account;
+    error.reset(make_account(
+                        CURRENCY_BITCOIN,
+                        "cQeGKosJjWPn9GkB7QmvmotmBbVg1hm8UjdN6yLXEWZ5HAcRwam7",
+                        reset_sp(account)));
+
+    PublicKeyPtr  pub_key_property =  account->get_public_key();
+
+    properties.bind_property("v", &binaty_data_4_property);
+
+    error.reset(properties_set_int32_value(&properties, "v", int_value));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(reference_value , *binaty_data_4_property);
+
+    error.reset(properties_set_string_value(&properties, "v", str.c_str()));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(reference_value , *binaty_data_4_property);
+
+    error.reset(properties_set_amount_value(&properties, "v", &amount));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(reference_value , *binaty_data_4_property);
+
+    error.reset(properties_set_public_key_value(&properties, "v",  pub_key_property.get()));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(reference_value , *binaty_data_4_property);
+}
+
+GTEST_TEST(PropertiesTestInvalidSetDate, properties_set_public_key_value)
```

Comment:
Rename test to (PropertiesTestInvalidType, BinaryData_property), same for test cases above.